README.md

```diff
@@ -0,0 +1,69 @@
+# OCP DNSNameResolver
+
+## Name
+
+OCP DNSNameResolver - updates the status of the [DNSNameResolver](https://github.com/openshift/api/blob/master/network/v1alpha1/types_dnsnameresolver.go)
+custom resources with IP addresses of matching resolved DNS names.
+
+## Description
+
+The spec of DNSNameResolver custom resource takes as input a DNS name. The DNS name can be either a regular or a wildcard DNS name. The plugin intercepts
+the DNS lookups of the matching DNS names and updates the status of these CRs with the IP addresses of the matching resolved names.
+
+The prerequisite for enabling this plugin are:
+- Adding the `DNSNameResolver` CRD to the Kubernetes API.
+- Adding `list` and `watch` permissions on the `DNSNameResolver` resources and `update` permission on the `DNSNameResolver/status` resource.
+
+NOTE: When adding the plugin to the `plugin.cfg` file in CoreDNS, care should be taken to place it before the plugins which will do the actual resolution of
+the DNS names that will be used in the DNSNameResolver custom resources. This will ensure that the plugin can intercept the DNS request and response in the
+plugin chain.
```

Comment:
Would be nice to name the concrete plugins to better steer the reader.

Reply:
Makes sense. Will add the plugin names.